### ISSUE TITLE
gdb.print() with conditional breakpoints giving inconsistent result.

### ISSUE THREAD

**Issue by roxx30198** [IntelPython/numba-dpex#1223]
```
(gdb) "side-by-side-2.py:19 if param_a == 5
No source file named side-by-side.py.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (side-by-side.py:17 if param_a==5) pending.
(gdb) r side-by-side.py --api=numba
(gdb) print param_c
```
The expected value for param_c is 15, however the print returns `param_c = 10`

**Comment by ZzEeKkAa** [comment]
Thank you for reporting it. Could you also include what is the value of `param_a` to determine either the issue is with value printed or with conditional breakpoint itself?

**Comment by ZzEeKkAa** [comment]
Also, if issues is reproducible with numba only and with regular gdb - check out if there is such issue in numba and if not - open the issue there.

**Comment by roxx30198** [comment]
Yeah, you are right, the issue description was misleading. `gdb.print()` gives inconsistent result, it works sometimes but fails sometimes as well and this is not related to conditional breakpoints.

**Comment by roxx30198** [comment]
> Also, if issues is reproducible with numba only and with regular gdb - check out if there is such issue in numba and if not - open the issue there.

For numba with latest gdb, everything works as expected and breakpoints are triggered, the issue is with numba-ndpx-kernel.

**Comment by ZzEeKkAa** [comment]
After investigating the issue, it appears that the issue was with calling python's `pexpect.expect()` to catch output from `info locals`. python function was called multiple times instead of calling once with the array input. After the first call buffer was read fully and second call was hanging up the program because expected output was already passed.
